packages/gatsby-plugin-image/src/components/layout-wrapper.tsx

```diff
@@ -24,9 +24,11 @@ if (hasNativeLazyLoadSupport) {
       mainImage.setAttribute('src', mainImage.dataset.src)
       mainImage.removeAttribute('data-src')
     }
-    if (mainImage.dataset.srcset) {
-      mainImage.setAttribute('srcset', mainImage.dataset.srcset)
-      mainImage.removeAttribute('data-srcset')
+
+    const sources = mainImage.parentNode.querySelectorAll('source[data-srcset]');
```

Comment:
Won't this miss the `img` `srcset`?

Reply:
```suggestion
    const sources = mainImage.parentNode.querySelectorAll('[data-srcset]');
```

Reply:
I think I shouldn't have removed the previous code. Both should exist. One will handle image and the other the additional sources.